content/post/debrick-core9g25-using-sam-ba-on-linux.md

```diff
@@ -0,0 +1,286 @@
+    erase::0x1000         erase from 0 to 0x1000
+
+* read - read from memory to a file
+Syntax:
+    read:<filename>:[<addr>]:[<length>]
+Examples:
+    read:firmware.bin              read all to firmware.bin
+    read:firmware.bin:0x1000       read from 0x1000 to end into firmware.bin
+    read:firmware.bin:0x1000:1024  read 1024 bytes from 0x1000 into firmware.bin
+    read:firmware.bin::1024        read 1024 bytes from start of memory into firmware.bin
+
+* verify - verify memory from a file
+Syntax:
+    verify:<filename>:[<addr>]
+Examples:
+    verify:firmware.bin         verify that start of memory matches firmware.bin
+    verify:firmware.bin:0x1000  verify that memory at offset 0x1000 matches firmware.bin
+
+* verifyboot - verify boot program from a file
+Syntax:
+    verifyboot:<filename>
+Example:
+    verify:firmware.bin  verify that start of memory matches boot program firmware.bin
+
+* write - write to memory from a file
+Syntax:
+    write:<filename>:[<addr>]
+Examples:
+    write:firmware.bin         write firmware.bin to start of memory
+    write:firmware.bin:0x1000  write firmware.bin at offset 0x1000
+
+* writeboot - write boot program to memory from a file
+Syntax:
+    writeboot:<filename>
+Example:
+    writeboot:firmware.bin  write firmware.bin as boot program at start of memory
+```
+
+## Serial connection
+Connect your Linux machine to the serial debug port of the device. I've used
+[this USB-to-serial cable][usb-to-serial].
+
+Before I continue I often use Minicom to validate if the serial connection is
+working. Start Minicom and configure the serial port with the settings below.
+You can open the settings menu by pressing `Ctrl-a` and then `o` (the letter,
+not the number).
+
+* baudrate: 115200
+* parity: none
+* bytesize: 8
+* stopbits: 1
+
+{{<figure src="/img/configure_minicom.png">}}
+
+Reboot the CPU. When you see human readable data in Minicom the serial port
+has been configured correctly. You probably should at least see the string
+"romboot".
+
+If you don't see anything, you probably need to switch your Rx and Tx lines.
+
+When Minicom prints garbage it's likely that the settings on both ends of the
+serial line doesn't match.
+
+## Boot from ROM
+If your serial connection has been set up it's time to fix debrick the NAND.
+The ROM of the AT91SAM9G25 contains a program which is also called SAM-BA,
+SAM-BA on your Linux machine interacts with SAM-BA on the ROM via the serial
+connection.
+
+In order to boot from ROM the BMS pin must be `1`. The BMS pin can be brought
+`1` by shortening the marked 2 island on the picture below.
+
+{{<figure src="/img/core9g25_with_marked_bms.jpg">}}
+
+Before you continue make sure that you've closed Minicom! Only 1 program at the
+time can use serial line.
+
+## Initialize RAM
+Enable the BMS pin while rebooting the device; the device will boot SAM-BA program
+that isBefore you continue make sure that you've closed Minicom! Only 1 program at the
```

Comment:
remove `Before you continue make sure that you've closed Minicom! Only 1 program at the time can use serial line.`